src/release.js

```diff
@@ -63,17 +66,13 @@ module.exports = function taoExtensionReleaseFactory(params = {}) {
     let gitClient;
     let githubClient;
 
+    if (!adaptees[subjectType]) {
+        throw new Error(`No implentation found for the type '${subjectType}'`);
+    }
     /**
      * @typedef adaptee - an instance of a supplemental API with methods specific to the release subject type
      */
-    let adaptee = {};
-
-    // Initialise the Adaptee and give it a copy of the release params and loaded data
-    if (subjectType === 'extension') {
-        adaptee = extensionApi(params, data);
-    } else if (subjectType === 'package' || subjectType === 'repo') {
-        adaptee = packageApi(params, data);
-    }
+    const adaptee = adaptees[subjectType](params);
```

Comment:
The adaptee is now instantiated without any `data` parameter being passed. Is this a deliberate choice?

If we don't want to pass `data` here any more (`setData()` is called at the end of `loadConfig`, after all), shouldn't the arguments of the 3 APIs also be changed?

Reply:
no I just forgot it